backend/models.py

```diff
@@ -1,23 +1,51 @@
 # Create your models here.
 from django.contrib.auth.models import AbstractUser
 from django.db import models
+from django.conf import settings
 
 import uuid
 
 
+class Host(models.Model):
+    url = models.URLField(max_length=400)
+    serviceAccountUsername = models.CharField(max_length=100, null=True, blank=True)
+    serviceAccountPassword = models.CharField(max_length=100, null=True, blank=True)
+
+
 class User(AbstractUser):
-    githubUrl = models.URLField(max_length=400)
+    githubUrl = models.URLField(max_length=400, blank=True)
 
+    # TODO do we need to store the host?
+    host = models.ForeignKey(
+        Host, null=True, blank=True, on_delete=models.CASCADE)
+
+    def get_full_user_id(self):
+        if self.host is None:
```

Comment:
When we create the user, can ensure that the host is always set so we don’t need to set it in this weird way here. Just like how we require the display name to be set, we can require the host to be set too.

Reply:
doable